### DISCUSSION TITLE
ApplicationModes not works in ContextualTabs?

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#34] [Q&A]
The solution code is attached.

I added a new 'Group2' in 'cmdTabGroupTableTools', and set the ApplicationModes='1'. And compile.

In code, what ever I set with _ribbon.SetModes methoed, the 'Group2' always visible.

[14-ContextualTabs.zip](https://github.com/user-attachments/files/16037666/14-ContextualTabs.zip)

BTW: Tested in Windows 11.

**Comment by harborsiem** [comment]
[See](https://learn.microsoft.com/en-us/windows/win32/windowsribbon/windowsribbon-controls-tabgroup)
